pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/DeleteAPIHybridClusterIntegrationTest.java

```diff
@@ -0,0 +1,247 @@
+    String pinotValue = firstAggregationResult.getString("value");
+    nRows = Integer.parseInt(pinotValue);
+  }
+
+  private int numRowsReturned() throws Exception{
+    org.json.JSONObject response = postQuery("select count(*) from 'mytable'");
+    org.json.JSONArray aggregationResultsArray = response.getJSONArray("aggregationResults");
+    org.json.JSONObject firstAggregationResult = aggregationResultsArray.getJSONObject(0);
+    String pinotValue = firstAggregationResult.getString("value");
+    return Integer.parseInt(pinotValue);
+  }
+
+  private void waitForTableReady() throws Exception {
+    while (true) {
+      int curOfflineRows = numRowsReturned();
+      if (curOfflineRows != nRows) {
+        Thread.sleep(1000);
+      }
+      else {
+        break;
+      }
+    }
+  }
+
+  // @Test Commenting this out for now because of HLC/LLC
+  public void deleteAllRealtimeSegmentsFromGetAPI() throws Exception{
+    sendGetRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).
+        forDeleteAllSegmentsWithTypeWithGetAPI(TABLE_NAME, "realtime"));
+    Thread.sleep(3000);
+
+    String postDeleteSegmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    Assert.assertEquals(getSegmentsFromJson(postDeleteSegmentList, "realtime").size(), 1);
+    Assert.assertNotSame(getSegmentsFromJson(postDeleteSegmentList, "offline").size(), 0);
+
+    repushRealtimeSegments();
+  }
+
+  @Test
+  public void deleteAllOfflineSegmentsFromGetAPI() throws Exception{
+    sendGetRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).
+        forDeleteAllSegmentsWithTypeWithGetAPI(TABLE_NAME, "offline"));
+    Thread.sleep(3000);
+
+    String postDeleteSegmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    Assert.assertEquals(getSegmentsFromJson(postDeleteSegmentList, "offline"), null);
+    Assert.assertNotSame(getSegmentsFromJson(postDeleteSegmentList, "realtime").size(), 0);
+
+    repushOfflineSegments();
+  }
+
+  @Override
+  public void testHardcodedQuerySet() {}
+
+  @Test
+  public void deleteOfflineSegmentFromGetAPI() throws Exception{
+    String segmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    JSONArray offlineSegmentsList = getSegmentsFromJson(segmentList, "offline");
+
+    sendGetRequest(ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).
+        forDeleteSegmentWithGetAPI(TABLE_NAME, offlineSegmentsList.get(0).toString(), "offline"));
+    Thread.sleep(5000);
+
+    String postDeleteSegmentList = sendGetRequest(
+        ControllerRequestURLBuilder.baseUrl(CONTROLLER_BASE_API_URL).forListAllSegments(TABLE_NAME));
+    JSONArray offlineSegmentsListReturn = getSegmentsFromJson(postDeleteSegmentList, "offline");
+    offlineSegmentsList.remove(0);
```

Comment:
Save the name of the segment you are deleting, and search to make sure that name is not there. Do not assume that segments will be returned in the same order all the time